rules.mk

```diff
@@ -459,12 +462,27 @@ define file_copy
 	$(CP) $(1) $(2)
 endef
 
+# POSIX usage of find with -mindepth and -maxdepth
+# $(1) => set of paths to recurse
+# $(2) => conditional statement(s)
+# $(3) => mindepth #
+# $(4) => maxdepth #
+# $(5) => eval expression (-exec)
+# $(6) => if set, do not follow links
+define find_depth
+	( \
+		for path in $(if $(word 2,$(1)),$(wildcard $(1)),$(1)); do \
+			$(FIND) $(if $(6),,-L) $$path '(' '!' -path "$(strip $$path$(call dir_depth,$(4)))" -o -prune ')' $(2) '(' '!' -path "$(if $(3),$(strip $$path$(call dir_depth,$(3))),*)" -o $(if $(5),$(5),-print) ')' ; \
+		done ; \
+	)
+endef
+
```

Comment:
`-mindepth` and `-maxdepth` might not be POSIX, but I'm pretty sure nearly all flavors of Unix added them.  While nominally correct, I disagree with this due to the uniform availability of the `find` operators.